includes/Core/Admin/Screens.php

```diff
@@ -288,14 +288,22 @@ private function get_screens() {
 						}
 					},
 					'render_callback'  => function( Context $context ) {
+						$setup_slug = $context->input()->filter( INPUT_GET, 'slug', FILTER_SANITIZE_STRING );
+						$reauth = $context->input()->filter( INPUT_GET, 'reAuth', FILTER_SANITIZE_STRING );
 						if ( $context->input()->filter( INPUT_GET, 'permaLink' ) ) {
 							?>
 							<div id="js-googlesitekit-dashboard-details" class="googlesitekit-page"></div>
 							<?php
 						} else {
-							?>
-							<div id="js-googlesitekit-dashboard" class="googlesitekit-page"></div>
-							<?php
+							if ( strlen( $setup_slug ) && strlen( $reauth ) ) {
+								?>
+								<div id="js-googlesitekit-dashboard" data-setup-module-slug=<?php echo wp_json_encode( $setup_slug ); ?> class="googlesitekit-page"></div>
+								<?php
+							} else {
+								?>
+								<div id="js-googlesitekit-dashboard" class="googlesitekit-page"></div>
+								<?php
+							}
```

Comment:
Let's avoid duplicating the HTML here – it's okay to define the data attribute with an empty string if there is no setup slug or `reAuth` wasn't requested in which case we can force it to be empty.

Also, is there a reason to use `wp_json_encode` here over `esc_attr`? It seems unnecessary for a plain string.

Reply:
> Also, is there a reason to use wp_json_encode here over esc_attr? It seems unnecessary for a plain string.

It was in the IB, wasn't sure for the reason there either, will update:
```output data-module-slug="<?php echo wp_json_encode( $module_info[ 'slug' ] ); ?>" as part of the #js-googlesitekit-module div.```

Reply:
https://github.com/google/site-kit-wp/pull/3805/commits/2f9128775aa2529d05e1fa7319b1a22032c2ab5b